src/search/BracketsSearchCursor.js

```diff
@@ -441,6 +464,9 @@ define(function (require, exports, module) {
         if (firstSearchResults.isGroupValuesEqual(0, secondSearchResults, secondSearchResults.itemCount() - 1)) {
```

Comment:
I tested this change, and it seems that comparing if they are equal is no longer required.  The test of enclosed covers both cases.